docs/src/pages/components/buttons/buttons.md

```diff
@@ -100,6 +100,12 @@ Here are some examples of customizing the component. You can learn more about th
 
 🎨 If you are looking for inspiration, you can check [MUI Treasury's customization examples](https://mui-treasury.com/styles/button).
 
+## Custom variants buttons
+
+Here are some examples of how you can add your custom variants to the component.
+
+{{"demo": "pages/components/buttons/CustomVariantButtons.js", "defaultCodeOpen": false}}
```

Comment:
I don't think that we should document it on a per-component basis. I think that we should make it something systematized. Maybe in https://material-ui.com/customization/components/#5-global-theme-variation